app/hooks/useSidebar.ts

```diff
@@ -0,0 +1,9 @@
+import React from "react";
+
+export const SidebarContext = React.createContext<React.Dispatch<
+  React.SetStateAction<number>
+> | null>(null);
+
+const useSidebar = () => React.useContext(SidebarContext);
```

Comment:
Using context is clearly the way to go but this particular implementation feels a little backward as the state of menus is stored inside of the sidebar component. Can we instead store the state in a hook here I wonder?

```suggestion

export function MenuContext() {
  const [openCount, setOpenCount] = React.useState(0);
  return <MenuContext.Provider value={{ openCount, setOpenCount }}>{children}</MenuContext.Provider>
}

const useMenuContext = () => React.useContext(SidebarContext);
```

You can now use the hook in both the sidebar and menu components:

```
const { openCount, setOpenCount } = useMenuContext();
```

I think this will work?